executor/index_lookup_hash_join.go

```diff
@@ -294,7 +394,14 @@ func (iw *indexHashJoinInnerWorker) run(ctx context.Context, cancelFunc context.
 			joinResult.err = task.err
 			break
 		}
-		err := iw.handleTask(ctx, cancelFunc, task, joinResult, h)
+		if task.keepOuterOrder {
+			if resultCh != nil {
+				close(resultCh)
+				resultCh = nil
+			}
+			resultCh = task.resultCh
+		}
+		err := iw.handleTask(ctx, cancelFunc, task, joinResult, h, resultCh)
```

Comment:
The count of the resouces `joinResult` is not matched to the in-flight tasks.
If the subsequent tasks always get the `joinResult`, a previous task may not get any `joinResult` and starve.

Reply:
No, every `innerWorker` holds there own `joinChkResourceCh` and the result chunk will only be re-sent to the corresponding `innerWorker`.